### ISSUE TITLE
Saunter breaks when you try register two same channels 

### ISSUE THREAD

**Issue by alan994** [tehmantra/saunter#133]
I have a simple scenario where in the same application I have publishing and subscribing code. Both operations are targeting the same topic (channel). When trying to generate a document I get an error. 

```
System.ArgumentException:
 An item with the same key has already been added.
  Key: conversation.user\r\n   
  at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)\r\n
   Saunter.Generation.DocumentGenerator.GenerateChannelsFromMethods(IEnumerable`1 asyncApiTypes, AsyncApiSchemaResolver schemaResolver, AsyncApiOptions options, JsonSchemaGenerator jsonSchemaGenerator, IServiceProvider serviceProvider)\r\n
   Saunter.Generation.DocumentGenerator.GenerateChannels(TypeInfo[] asyncApiTypes, AsyncApiSchemaResolver schemaResolver, AsyncApiOptions options, JsonSchemaGenerator jsonSchemaGenerator, IServiceProvider serviceProvider)\r\n
   Saunter.Generation.DocumentGenerator.GenerateDocument(TypeInfo[] asyncApiTypes, AsyncApiOptions options, AsyncApiDocument prototype, IServiceProvider serviceProvider)\r\n
   Saunter.Generation.AsyncApiDocumentProvider.GetDocument(AsyncApiOptions options, AsyncApiDocument prototype)\r\n
   Saunter.AsyncApiMiddleware.Invoke(HttpContext context)\r\n
   Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)\r\n
   Microsoft.AspNetCore.HttpLogging.HttpLoggingMiddleware.InvokeInternal(HttpContext context)\r\n
....
```

The problem is [here](https://github.com/tehmantra/saunter/blob/25ca43e38cd002a812f055517c794dc062344668/src/Saunter/Generation/DocumentGenerator.cs#L82). Underlaying collection is `IDictionary` and, if I understood correctly, it is not possible to have the same channels declared multiple times in the same application. Are there any plans to support this scenario?

**Comment by alan994** [comment]
@tehmantra is this project still active?

**Comment by tehmantra** [comment]
@alan994 sorry have been on Xmas holidays. I Will get to this in the coming week 🙂

**Comment by alan994** [comment]
If I can help somehow, let me know. Tnx.

**Comment by rkpin** [comment]
we have a couple of scenarios where we tried to register more than one operation with the same channel.   
1. app publishing and subscribing from the same channel/ topic.
2. app with multiple subscriptions from same channel/ topic. 

saunter breaks if you have multiple channel attributes with the same name. AsychAPI 2.3 spec supports more than one subscription and publishers under a single channel. It would be great if saunter also supports this.

**Comment by alan994** [comment]
@tehmantra Did you have a chance to look at this? Any open questions around this issue?

**Comment by cmcjcharters** [comment]
I have also encountered this issue, and am happy to propose a PR if I can get something to work. We also have a case where we have pub/sub on the same channel (Websockets)

**Comment by MohamadTahir** [comment]
Any progress on this?

**Comment by tehmantra** [comment]
Fixed in v0.10.0 https://www.nuget.org/packages/Saunter/0.10.0
